src/main/java/com/google/api/codegen/transformer/HeaderRequestParamTransformer.java

```diff
@@ -0,0 +1,80 @@
+/* Copyright 2017 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.config.GapicMethodConfig;
+import com.google.api.codegen.config.ProtoField;
+import com.google.api.codegen.config.TransportProtocol;
+import com.google.api.codegen.viewmodel.HeaderRequestParamView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.MessageType;
+import com.google.api.tools.framework.model.Method;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+
+public class HeaderRequestParamTransformer {
+  public List<HeaderRequestParamView> generateHeaderRequestParams(MethodContext context) {
```

Comment:
These can be static.

Reply:
They haven't been for the other transformers.